apps/remix-ide/src/app/tabs/hardhat-provider.js

```diff
@@ -41,21 +43,14 @@ export default class HardhatProvider extends Plugin {
 
   sendAsync (data) {
     return new Promise((resolve, reject) => {
+      if (this.blocked) return reject(new Error('provider temporarily blocked'))
```

Comment:
Error message can be updates as `blocked` can be concerning for users. may be something like `provider unable to connect`